include/vcpkg/base/checks.h

```diff
@@ -32,6 +40,18 @@ namespace vcpkg::Checks
                               Strings::format(error_message_template, error_message_arg1, error_message_args...));
         }
     }
+    template<class Conditional, class Message, class... Args>
+    typename Message::is_message_type check_exit(const LineInfo& line_info,
```

Comment:
Speaking of code size; maybe this is a stupid question, but why is `Conditional` not `bool`?

Reply:
to allow passing in, say, `std::unique_ptr`, since the conversion is explicit.

Reply:
How often do we do that and is the resulting code size penalty worthwhile?